src/simulation.hpp

```diff
@@ -1394,24 +1576,61 @@ auto AMRSimulation<problem_t>::PlotFileName(int lev) const -> std::string {
   return amrex::Concatenate(plot_file, lev, 5);
 }
 
+template <typename problem_t>
+void AMRSimulation<problem_t>::AverageFCToCC(
+    amrex::MultiFab &mf_cc, const amrex::MultiFab &mf_fc, int dstcomp_start,
+    int srccomp_start, int srccomp_total, int nGrow) const {
+  // itterate over the domain
+  auto const &state_cc = mf_cc.arrays();
+  auto const &state_fc = mf_fc.const_arrays();
+  amrex::ParallelFor(
+      mf_cc, amrex::IntVect(AMREX_D_DECL(nGrow, nGrow, nGrow)),
+      [=] AMREX_GPU_DEVICE(int boxidx, int i, int j, int k) {
+        for (int icomp = 0; icomp < srccomp_total; ++icomp) {
+          state_cc[boxidx](i, j, k, dstcomp_start + icomp) =
+              0.5 * (state_fc[boxidx](i, j, k, srccomp_start + icomp) +
+                     state_fc[boxidx](i + 1, j, k, srccomp_start + icomp));
+          if constexpr (AMREX_SPACEDIM > 1) {
+            state_cc[boxidx](i, j, k, dstcomp_start + icomp) =
```

Comment:
This does not make sense. You cannot average a vector into a scalar. You need as many cell-centered components as face-centered components. Only the centering is different, not the number of components.